pkg/tcpip/buffer/prependable.go

```diff
@@ -66,7 +69,12 @@ func (p *Prependable) TrimBack(size int) {
 // slice that represents the reserved space.
 func (p *Prependable) Prepend(size int) []byte {
 	if size > p.usedIdx {
-		return nil
```

Comment:
Can Prepend() return nil? It looks like nobody checks the return value. 
from fdbased.go:
```golang

// WritePacket writes outbound packets to the file descriptor. If it is not
// currently writable, the packet is dropped.
func (e *endpoint) WritePacket(r *stack.Route, gso *stack.GSO, hdr buffer.Prependable, payload buffer.VectorisedView, protocol tcpip.NetworkProtocolNumber) *tcpip.Error {
	if e.hdrSize > 0 {
		// Add ethernet header if needed.
		eth := header.Ethernet(hdr.Prepend(header.EthernetMinimumSize))
		ethHdr := &header.EthernetFields{
			DstAddr: r.RemoteLinkAddress,
			Type:    protocol,
		}

		// Preserve the src address if it's set in the route.
		if r.LocalLinkAddress != "" {
			ethHdr.SrcAddr = r.LocalLinkAddress
		} else {
			ethHdr.SrcAddr = e.addr
		}
		eth.Encode(ethHdr)
	}

```

Reply:
You're intuitions are likely correct. You're intentions for the change look good.

Reply:
I would rather not change this.

1. I am concerned that this is going to mask bugs.
1. This is not a particularly efficient way to do slice extension as it allocates every time. The point of the prependable buffer is that is makes prepending efficient. The simple way to do prepending is `append(newData, oldData...)`, an option still available for users who don't want to bother with reserving space up-front.
1. I am concerned that this is going to make it too easy to introduce major performance regressions.

Reply:
reverted back